README.md

```diff
@@ -1639,19 +1639,35 @@ Repository: https://github.com/aquasecurity/trivy-ci-test
 
 ### GitLab CI using Trivy container
 
+To scan a previously built image that has already been pushed into the
+GitLab container registry the following CI job manifest can be used.
+Note that `entrypoint` needs to be unset for the `script` section to work.
+In case of a non-public GitLab project Trivy additionally needs to
+authenticate to the registry to be able to pull your application image.
+Finally, it is not necessary to clone the project repo as we only work
+with the container image.
+
 ```yaml
 container_scanning:
-  image:                           docker.io/aquasec/trivy:latest
+  image:
+    name: docker.io/aquasec/trivy:latest
+    entrypoint: [""]
```

Comment:
I only recently started to play around with Trivy so I do not know since when the `ENTRYPOINT ["trivy"]` is in the Dockerfile.  However, this means that there is no shell so the commands from the `script` section will not be executed.  Instead they are interpreted as command line arguments to the `trivy` command itself.

You can reproduce the error by e.g. doing something like:

```
$ docker run --rm  -v $HOME/data/trivycache/:/root/.cache/ aquasec/trivy echo "Hello" 
2021-02-15T11:25:56.222Z        ERROR   multiple targets cannot be specified
2021/02/15 11:25:56 failed to initialize options: arguments error
```

vs.

```
$ docker run --rm --entrypoint "" -v $HOME/data/trivycache/:/root/.cache/ aquasec/trivy  echo "Hello"
Hello
```

Reply:
Oh, and the problem authenticating to the GitLab registry only happens for non-public projects.  Pulling an image from a public project does not require authentication, I assume (not tested).  But it does not hurt, either.

Reply:
The repro would prove your point, but that holds only when reproing on a bare docker. Gitlab's docker-runner overrides the entrypoint of everything that's mentioned in `image:` by default, in order to actually get into the container and run the script inside. But, in its' own way. So maybe your edit here is cleaner. But that would be necessary only if we'd use trivy image as a `service:`.

Reply:
Yeah, thanks for letting know about there are those variables to be used with the non-open registries!

Reply:
> The repro would prove your point, but that holds only when reproing on a bare docker. Gitlab's docker-runner overrides the entrypoint of everything that's mentioned in `image:` by default, in order to actually get into the container and run the script inside.  [...]

Well, in fact I had the exact same error when running under Gitlab's Docker runner.  That was the reason why I started digging into this in the first place. Here is how it looked in the Gitlab job log:

```
Executing "step_script" stage of the job script
00:00
2021-02-13T01:10:26.289Z	ERROR	multiple targets cannot be specified
2021/02/13 01:10:26 failed to initialize options: arguments error
Cleaning up file based variables
ERROR: Job failed: exit code 1
```

Reply:
>  Gitlab's docker-runner overrides the entrypoint of everything that's mentioned in `image:` by default …

Now this made me curious so I searched the docs a bit more.  The Gitlab runner does **not** do that.  Quite the contrary!  [The documentation](https://docs.gitlab.com/ee/ci/docker/using_docker_images.html#overriding-the-entrypoint-of-an-image) says:

> 1. The runner starts a Docker container using the defined entrypoint (default from Dockerfile that may be overridden in .gitlab-ci.yml)
> 1. The runner attaches itself to a running container.
>
> [...]
>
> To override the entrypoint of a Docker image, you should define an empty entrypoint in .gitlab-ci.yml, so the runner does not start a useless shell layer.

Reply:
TIL. Thank you!